src/RWA007-A/scripts/RWA007Deployment.s.sol

```diff
@@ -0,0 +1,174 @@
+        NAME                      = getEnvStringRequired("NAME");
+        SYMBOL                    = getEnvStringRequired("SYMBOL");
+        LETTER                    = getEnvStringRequired("LETTER");
+        SYMBOL_LETTER             = string(abi.encodePacked(SYMBOL, "_", LETTER));
+
+        ILK                       = bytesToBytes32(abi.encodePacked(SYMBOL, string("-"), LETTER));
+
+    }
+
+    function run() external {
+        vm.startBroadcast();
+
+        // tokenize it
+        address RWA_TOKEN = getEnvAddress("RWA_TOKEN");
+        if (RWA_TOKEN == address(0)) {
+            RWA_TOKEN = address(RWA_TOKEN_FAB.createRwaToken(NAME, SYMBOL, MCD_PAUSE_PROXY));
+        }
+
+        // join it
+        address RWA_JOIN = getEnvAddress("RWA_JOIN");
+        if (RWA_JOIN == address(0)) {
+            RWA_JOIN =  JOIN_FAB.newAuthGemJoin(MCD_PAUSE_PROXY, ILK, RWA_TOKEN);
+        }
+
+        // route it
+        address RWA_OUTPUT_CONDUIT = getEnvAddress("RWA_OUTPUT_CONDUIT");
+        if (RWA_OUTPUT_CONDUIT == address(0)) {
+            RwaOutputConduit3 outputC =  new RwaOutputConduit3(MCD_PSM_USDC_A);
+            outputC.rely(MCD_PAUSE_PROXY);
+
+            RWA_OUTPUT_CONDUIT = address(outputC);
+        }
+        
+        // urn it
+        address RWA_URN = getEnvAddress("RWA_URN");
+        if (RWA_URN == address(0)) {
+            RwaUrn2 urn = new RwaUrn2(MCD_VAT, MCD_JUG, RWA_JOIN, MCD_JOIN_DAI, RWA_OUTPUT_CONDUIT);
+            urn.rely(MCD_PAUSE_PROXY);
+            urn.deny(msg.sender);
+
+            RWA_URN = address(urn);
+
+            // Set _quitTo address to the URN and deny deplyer
+            RwaOutputConduit3(RWA_OUTPUT_CONDUIT).file("quitTo", RWA_URN);
+            RwaOutputConduit3(RWA_OUTPUT_CONDUIT).deny(msg.sender);
+        }
+
+        // jar it
+        address RWA_JAR = getEnvAddress("RWA_JAR");
+        if (RWA_JAR == address(0)) {
+            RWA_JAR = address(new RwaJar(address(CHANGELOG)));
+        }
+
+        // route it JAR
+        address RWA_INPUT_CONDUIT_JAR = getEnvAddress("RWA_INPUT_CONDUIT_JAR");
+        if (RWA_INPUT_CONDUIT_JAR == address(0)) {
+            RwaInputConduit3 inputCJar = new RwaInputConduit3(MCD_PSM_USDC_A, RWA_JAR);
+            inputCJar.rely(MCD_PAUSE_PROXY);
+            inputCJar.deny(msg.sender);
+
+            RWA_INPUT_CONDUIT_JAR = address(inputCJar);
+        }
+
+        // route it URN
+        address RWA_INPUT_CONDUIT_URN = getEnvAddress("RWA_INPUT_CONDUIT_URN");
+        if (RWA_INPUT_CONDUIT_URN == address(0)) {
+            RwaInputConduit3 inputCUrn = new RwaInputConduit3(MCD_PSM_USDC_A, RWA_URN);
+            inputCUrn.rely(MCD_PAUSE_PROXY);
+            inputCUrn.deny(msg.sender);
+
+            RWA_INPUT_CONDUIT_URN = address(inputCUrn);
+        }
+        
+        vm.stopBroadcast();
+
+        logJSONTuple("SYMBOL", SYMBOL);
+        logJSONTuple("NAME", NAME);
+        logJSONTuple("ILK", string(abi.encodePacked(SYMBOL, string("-"), LETTER)));
```

Comment:
You already have the `ILK` variable. No need to recalculate it here:
```suggestion
        logJSONTuple("ILK", ILK);
```

Reply:
ILK is bytes32 wich is not what we expect here

Reply:
Can't you use the `bytes32ToString()` helper method then?

Reply:
Done

Reply:
I might be wrong, but I think you can convert `bytes32` to a `string` simply with:
```solidity
string(abi.encodePacked(bytes32))
```

Reply:
The idea is to remove the noise from the logging part to it's easier to see what's being logged.

Reply:
`string(abi.encodePacked(ILK))` will end up with string representation of bytes "0x....". I try this way before, so it's not an easy way to convert bytes32 -> bytes.  I can just create another constant for ILK_STRING or leave it as it is.

Reply:
I end up with this ` "ILK": "RWA007AT2-A\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000"`, not sure where this zeros coming from (probably something related to bytes32 -> bytes)

Reply:
Yeah, apparently you have to use this helper function:

```solidity
function bytes32ToString(bytes32 src) internal pure returns (string memory result) {
    uint8 length = 0;
    while (src[length] != 0 && length < 32) {
        length++;
    }
    assembly {
        result := mload(0x40)
        // new "memory end" including padding (the string isn't larger than 32 bytes)
        mstore(0x40, add(result, 0x40))
        // store length in memory
        mstore(result, length)
        // write actual data
        mstore(add(result, 0x20), source)
    }
}
```

Reply:
There's also this version, which is slightly less optimized, but doesn't really matter in this case:

```solidity
function bytes32ToString(bytes32 src) internal pure returns (string memory) {
  bytes memory dst = new bytes(32);
  for (uint256 i; i < 32; i++) {
    if (src[i] == 0) break;
    dst[i] = src[i];
  }
  return string(dst);
}
```

Reply:
Good, will fix it soon

Reply:
Done